etna/blog/models.py

```diff
@@ -0,0 +1,130 @@
+from django.utils.functional import cached_property
+
+from wagtail.admin.panels import FieldPanel
+from wagtail.api import APIField
+from wagtail.fields import StreamField
+
+from etna.core.models import (
+    BasePageWithRequiredIntro,
+    ContentWarningMixin,
+    HeroImageMixin,
+)
+from etna.core.serializers import DefaultPageSerializer
+from etna.people.models import AuthorPageMixin
+
+from .blocks import BlogPostPageStreamBlock
+
+
+class BlogIndexPage(HeroImageMixin, BasePageWithRequiredIntro):
+    """Blog index page
+
+    This is the parent page for all blog posts. It is used to
+    display a list of blog posts, and blog pages.
+    """
+
+    subpage_types = ["blog.BlogPostPage", "blog.BlogPage"]  # TBC
+
+    parent_page_types = ["home.HomePage"]  # TBC
+
+    content_panels = (
+        BasePageWithRequiredIntro.content_panels + HeroImageMixin.content_panels
+    )
+
+    promote_panels = BasePageWithRequiredIntro.promote_panels
+
+    @cached_property
+    def blog_pages(self):
+        return (
+            BlogPage.objects.all()
+            .order_by("-first_published_at")
```

Comment:
I think we need to order these by a custom date field (see comment in `BlogPostPage` below)

Reply:
Sorry, I meant to remove this one here - I think the blogs should be ordered alphabetically, and the blog posts should be ordered chronologically

Reply:
That sounds like a good plan 👍🏼